examples/UNet/generators.py

```diff
@@ -1,21 +1,30 @@
 __author__ = 'Fabian Isensee'
 import numpy as np
 
-def batch_generator(data, target, BATCH_SIZE):
+def batch_generator(data, target, BATCH_SIZE, shuffle=False):
     '''
     just a simple batch iterator, no cropping, no rotation, no anything
     '''
     np.random.seed()
     idx = np.arange(data.shape[0])
+    if shuffle:
+        np.random.shuffle()
+    idx_2 = np.array(idx)
+    # if BATCH_SIZE is larger than len(data) we need to artificially enlarge the idx array (loop around)
+    while BATCH_SIZE > len(idx):
+        idx_2 = np.concatenate((idx_2, idx))
+    del(idx)
     while True:
-        ids = np.random.choice(idx, BATCH_SIZE)
-        yield np.array(data[ids]), np.array(target[ids])
+        ctr = 0
+        yield np.array(data[idx_2[ctr:ctr+BATCH_SIZE]]), np.array(target[idx_2[ctr:ctr+BATCH_SIZE]])
+        ctr += BATCH_SIZE
+        if ctr >= data.shape[0]:
+            ctr -= data.shape[0]
```

Comment:
Sorry, this looks overly complicated. What about:
```python
def batch_generator(data, target, BATCH_SIZE, shuffle=False):
    if shuffle:
        while True:
            ids = np.random.choice(len(data))
            yield data[ids], target[ids]
    else:
        for idx in range(len(data) // BATCH_SIZE):
            ids = slice(idx * BATCH_SIZE, (idx + 1) * BATCH_SIZE)
            yield data[ids], target[ids]
```
This will discard the last incomplete batch. If you want to include it, you just need to change the range to get up to `np.ceil(len(data) / float(BATCH_SIZE))`.